gocourse07/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"context"
+	"math/rand/v2"
+	"sync"
+	"time"
+)
+
+func generateLightSensors(n int) []Sensor {
+	lightSensors := make([]Sensor, n)
+	for i := range lightSensors {
+		lightSensors[i] = LightSensor{
+			ID:    i,
+			Light: rand.IntN(100),
+		}
+	}
+	return lightSensors
+}
+
+func generateWetnessSensors(n int) []Sensor {
+	wetnessSensors := make([]Sensor, n)
+	for i := range wetnessSensors {
+		wetnessSensors[i] = WetnessSensor{
+			ID:      i,
+			Wetness: rand.IntN(100),
+		}
+	}
+	return wetnessSensors
+}
+
+func generateTemperatureSensors(n int) []Sensor {
+	temperatureSensors := make([]Sensor, n)
+	for i := range temperatureSensors {
+		temperatureSensors[i] = TemperatureSensor{
+			ID:          i,
+			Temperature: rand.IntN(100),
+		}
+	}
+	return temperatureSensors
+}
+
+func main() {
+
+	deadline := time.Now().Add(100 * time.Millisecond)
+	ctx, cancel := context.WithDeadline(context.Background(), deadline)
+	defer cancel()
+
+	dataChannel := make(chan Data)
```

Comment:
Напевно, можемо зробити канал буферизованим:
```suggestion
	dataChannel := make(chan Data, 4)
```

https://go.dev/doc/effective_go#channels